src/quest/manager.py

```diff
@@ -66,6 +68,7 @@ def _get_workflow(self, workflow_id: str):
         return self._workflows[workflow_id]
 
     def _remove_workflow(self, workflow_id: str):
+        self._workflows.get(workflow_id)._history.clear()
```

Comment:
Hacky. Haven't tested this yet. Didn't seem like _remove_workflow is always called after a workflow is finished? Guessing_game didn't seem to run this.